dek-registry-client/src/main/java/io/confluent/dekregistry/client/CachedDekRegistryClient.java

```diff
@@ -118,16 +125,17 @@ public List<String> listDeks(String kekName, boolean lookupDeleted)
   }
 
   @Override
-  public Dek getDek(String kekName, String subject, boolean lookupDeleted)
+  public List<Integer> listDekVersions(String kekName, String subject,
+      DekFormat algorithm, boolean lookupDeleted)
       throws IOException, RestClientException {
-    return getDek(kekName, subject, null, lookupDeleted);
+    return restService.listDekVersions(kekName, subject, algorithm, lookupDeleted);
   }
 
   @Override
   public Dek getDek(String kekName, String subject, DekFormat algorithm, boolean lookupDeleted)
       throws IOException, RestClientException {
     try {
-      return dekCache.get(new DekId(kekName, subject, algorithm, lookupDeleted), () ->
+      return dekCache.get(new DekId(kekName, subject, null, algorithm, lookupDeleted), () ->
```

Comment:
whats the difference between version `null` and `-1`?

Reply:
-1 is the latest version, this is the same as for subject versions.  I'll add a constant for it.

null means that the DEK is not using rotation, in this case it will default to MIN_VERSION, which is 1